### PR TITLE
Fix examples and documentations

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: examples/stat_object.py [hunk 1]
```diff
@@ -36,7 +33,7 @@
 
 # Get object information of version-ID.
 result = client.stat_object(
-    "my-bucketname", "my-objectname",
+    "my-bucket", "my-object",
```
Comment (ebozduman) [inline examples/stat_object.py:36]: They are all in comments, but there are still 33 references to `my-bucketname` and `my-objectname` in 21 files.
If you'd like, you may want to change them all to `my-bucket` and to `my-object` for consistency purposes.

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: docs/API.md [hunk 2]
```diff
@@ -90,56 +82,40 @@ __Parameters__
...
-
-minioClient = Minio(
-	'play.min.io',
-	access_key='Q3AM3UQ867SPQQA43P2F',
-	secret_key='zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG',
-)
-```
 
-> NOTE: If there is a corporate proxy, specify a custom httpClient using *urllib3.ProxyManager* as shown below:
+# Create client with anonymous access.
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply docs/API.md:32]: Restructured the doc.

File: docs/API.md [hunk 3]
```diff
@@ -252,35 +232,35 @@ __Example__
...
+for obj in objects:
+    print(obj)
 
 # List objects information recursively after object name
-# 'hello/world/1'.
-objects = minio.list_objects(
-    'foo', recursive=True, start_after='hello/world/1',
+# "my/prefix/world/1".
+objects = client.list_objects(
+    "my-bucket", recursive=True, start_after="my/prefix/world/1",
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply docs/API.md:239]: Done

File: docs/API.md [hunk 4]
```diff
@@ -431,12 +411,13 @@ __Parameters__
 | `events`      | _list_ | Events to listen.                           |
 
 ```py
-iter = minio.listen_bucket_notification(
-    "my-bucketname",
-    events=('s3:ObjectCreated:*', 's3:ObjectAccessed:*'),
+events = client.listen_bucket_notification(
+    "my-bucket",
+    prefix="my-prefix/",
+    events=["s3:ObjectCreated:*", "s3:ObjectRemoved:*"],
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply docs/API.md:396]: Done

File: examples/stat_object.py (hunk 1 above)
Comment (balamurugana) [reply examples/stat_object.py:36]: I don't find files containing those. Can you list those files?

Comment (ebozduman) [reply examples/stat_object.py:36]: Selection_025

Comment (balamurugana) [reply examples/stat_object.py:36]: None of them except `api.py` are in the repository. You would need to tidy up your work directory.

File: docs/API.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ebozduman) [reply docs/API.md:239]: We also need  these new parameters in the title, right?

[the remark this answers is not in the document]
Comment (balamurugana) [reply docs/API.md:239]: Fixed

**Review by ebozduman — APPROVED** [review]
LGTM

**Review by ebozduman — APPROVED** [review]
LGTM

File: minio/api.py [hunk 5]
```diff
@@ -100,12 +94,21 @@ class Minio:  # pylint: disable=too-many-public-methods
...
+        # Create client with access key and secret key with specific region.
+        client = Minio(
+            "play.minio.io:9000",
+            access_key="Q3AM3UQ867SPQQA43P2F",
+            secret_key="zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG",
+            region="my-region",
+        )
 
-    **NOTE on concurrent usage:** The `Minio` object is thread safe when using
+    **NOTE on concurrent usage:** `Minio` object is thread safe when using
```
Comment (ravindk89) [inline minio/api.py:111]: If my wording from [API.md](https://github.com/minio/minio-py/pull/1038/files#diff-b57590968a12cee85a37c1b91d8cc7092cd8b68e50b242c39a178121db82a797R25) looks OK, use the same wording here.

**Review by ravindk89 — CHANGES_REQUESTED** [review]
Strictly speaking, most of my feedback are nits and are closer to being "polish" than something that is absolutely necessary. If the priority is to get this out, I don't think anything here would be a hard block to user success.

File: README.md [hunk 6]
```diff
@@ -1,211 +1,96 @@
...
+    found = client.bucket_exists("asiatrip")
+    if not found:
+        client.make_bucket("asiatrip")
+    else:
+        print("Bucket 'asiatrip' already exists")
+
+    # Upload '/home/user/Photos/asiaphotos.zip' as object name
+    # 'asiaphotos-2015.zip' to bucket 'asiatrip'.
+    client.put_object(
+        "asiatrip", "asiaphotos-2015.zip", "/home/user/Photos/asiaphotos.zip",
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply README.md:62]: No. That's not the scope the example. We just need to give an intro to the SDK by an example.

File: docs/API.md [hunk 7]
```diff
@@ -1,32 +1,65 @@
...
-minioClient = Minio(
-	'play.min.io',
-	access_key='Q3AM3UQ867SPQQA43P2F',
-	secret_key='zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG',
-	secure=True,
-)
-```
 
-## AWS S3
+**NOTE on concurrent usage:** `Minio` object is thread safe when using the Python `threading` library. Specifically, it is **NOT** safe to share it between multiple processes, for example when using `multiprocessing.Pool`. The solution is simply to create a new `Minio` object in each process, and not share it between processes.
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply docs/API.md:25]: As this is existing text in the doc, I am keeping it as is.

File: docs/API.md [hunk 8]
```diff
@@ -1,32 +1,65 @@
...
+
+__Example__
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply docs/API.md:27]: I think we need to provide a heading to example code; it is not based on the code.

I don't think users just copy (why?) and use it. It is just a reference code how to use in various cases.

File: docs/API.md [hunk 9]
```diff
@@ -224,63 +180,66 @@ __Parameters__
...
+| `use_api_v1`        | _bool_ | Flag to control to use ListObjectV1 S3 API or not.       |
 
 __Return Value__
 
-| Return                                                    |
-|:----------------------------------------------------------|
-| An iterator contains object information as _minio.Object_ |
+| Return                  |
+|:------------------------|
+| An iterator of _Object_ |
```
[the remark this answers is not in the document]
Comment (balamurugana) [reply docs/API.md:208]: `listObjects()` returns iterator not `list`

File: docs/API.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ravindk89) [reply docs/API.md:27]: Personal experience - previously have had issues of users doing blind copy/paste and then reporting issues back. If we haven't run into much of that, then there's no reason to deal with it right this instant.

**Review by ravindk89 — APPROVED** [review]

**Review by ebozduman — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**